.ci/Jenkinsfile

```diff
@@ -0,0 +1,94 @@
+#!groovy
+/**
+ * Jenkins pipeline to build Corda DJVM
+ */
+
+/**
+ * Kill already started job.
+ * Assume new commit takes precendence and results from previous
+ * unfinished builds are not required.
+ * This feature doesn't play well with disableConcurrentBuilds() option
+ */
+@Library('existing-build-control')
+import static com.r3.build.BuildControl.killAllExistingBuildsForJob
+killAllExistingBuildsForJob(env.JOB_NAME, env.BUILD_NUMBER.toInteger())
+
+/**
+ * Sense environment
+ */
+boolean isPR = (env.BRANCH_NAME.startsWith('PR'))
+boolean isRelease = (env.TAG_NAME =~ /^release\/.+$/ || env.BRANCH_NAME =~ /^release\/.+$/)
```

Comment:
These checks are important for publishing to Artifactory.
They need to tell if this is a build to publish a snapshot, or a build to publish a release.
Please check with @chrisr3 about the details. Specifically:

* build for which branch or branches should publish a DJVM snapshot to Artifactory: it looks to me it is just the `master` branch
* builds for which branch or branches or tags should publish a release to Artifactory: it looks to me it is just tags matching `release/*` pattern.

Reply:
Sounds OK. I have been publishing releases to `corda-dependencies` and SNAPSHOTs to `corda-dependencies-dev`. And I have kept the `release/1.0` and `release/1.1` branches just in case I have needed to create an emergency `1.1.x` release.

Reply:
@chrisr3 please note, due to the way Jenkins scans repositories and how we configured Jenkins for DJVM at the moment, it is not possible to build from a tag, but *skip* builds for `release/x` branch. I guess it is still fine, as it will act as a prerelease build.